src/advanced/blockchain-explorer.yaml

```diff
@@ -0,0 +1,528 @@
+openapi: 3.0.0
+info:
+  title: Explorer API endpoints
+  version: 0.9.0
+  description: >
+    All explorer endpoints are public.
+    `enable_blockchain_explorer` local configuration parameter
+    allows to turn explorer endpoints on/off.
+
+servers:
+- url: http://example.com/api/explorer/v1
+
+tags:
+- name: Read Requests
+  description: Public read-only API available to everybody.
+
+paths:
+  /transactions:
+    get:
+      tags: [Read Requests]
+      summary: Get transaction by hash
+      description: >
+        Searches for a transaction, either committed or uncommitted, by the hash.
+      parameters:
+      - in: query
+        name: hash
+        required: true
+        description: The hash of the transaction to be searched.
+        schema:
+          $ref: "#/components/schemas/Hash"
+      responses:
+        200:
+          description: >
+            Known committed or uncommitted transaction on the blockchain.
+
+            Uncommitted transactions are returned as `SerializedTransaction` objects
+            with additional **type** field set to `in-pool`.
+
+            Known transactions are returned as `KnownTransaction` objects
+            with **type** field set to `committed`.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/KnownTransaction"
+        404:
+          description: >
+            Unknown transaction.
+            Response contains only **type** field set to `unknown`.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/UnknownTransaction"
+
+  /block:
+    get:
+      tags: [Read Requests]
+      summary: Get block by height
+      description: >
+        Returns the content for a block of a specific height.
+      parameters:
+      - in: query
+        name: height
+        required: true
+        description: The height of the desired block.
+        schema:
+          type: integer
+          example: 20
+      responses:
+        200:
+          description: >
+            A JSON object describing the block.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  block:
+                    $ref: "#/components/schemas/BlockHeader"
+                  precommits:
+                    type: array
+                    description: The list of `Precommit` messages supporting the block.
+                    items:
+                      $ref: "#/components/schemas/Precommit"
+                  txs:
+                    type: array
+                    description: The list of the transactions included into the block.
+                    items:
+                      $ref: "#/components/schemas/SerializedTransaction"
```

Comment:
I think the transaction hashes are used here.

Reply:
Yeah, my bad. Thanks for having hawk eyes.

This object is actually serialized from this struct:

```rust
pub struct BlockInfo {
    pub block: Block,
    pub precommits: Vec<Precommit>,
    pub txs: Vec<Hash>,
}
```